### DISCUSSION TITLE
Trying to convert SpringBoot to quarkus 3

### DISCUSSION THREAD

**Discussion by three-old-coders** [quarkusio/quarkus#40509] [Q&A]
Hi! I am trying to build a native quarkus app based on a springboot WEB Application (using @Controller NOT @RestController). Unfortunately the application starts without issues, quarkus listens on 0.0.0.0:8080 but nothing happens, no error, no response. When reading the docs quarkus always refers to @RestController, so is it really impossible to create a SpringBoot based Web Application (based on Tymeleaf or Rocker or whatever)?

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
In Quarkus, you would likely use [Quarkus REST and Qute](https://quarkus.io/guides/qute#hello-qute-rest). 
Another option would be to use [Renarde](https://docs.quarkiverse.io/quarkus-renarde/dev/index.html).

**Reply by three-old-coders** [reply]
Many thanks. Will have a look at this. 
The drawback is all controller classes need to be rewritten or require to duplicate the annotations. How will Qute/Renarde interact with spring IOC as the spring is not aware of this class ignored @Controller annotate class and all @Autowired and method parameter will fail?

**Reply by geoand** [reply]
My point above was that you would need to use Quarkus specific APIs, not Spring ones.
